web/src/unread_ui.js

```diff
@@ -81,11 +82,10 @@ export function update_unread_counts(skip_animations = false) {
     // Side effects from here down:
     // This updates some DOM elements directly, so try to
     // avoid excessive calls to this.
-    activity.update_dom_with_unread_counts(res);
-    top_left_corner.update_dom_with_unread_counts(res, skip_animations);
-    stream_list.update_dom_with_unread_counts(res);
-    pm_list.update_dom_with_unread_counts(res);
-    topic_list.update();
+    // See `ui_init.initialize_unread_ui` for the registered hooks.
+    for (const hook of update_unread_counts_hooks) {
+        hook(res, skip_animations);
+    }
     const notifiable_unread_count = unread.calculate_notifiable_count(res);
     notifications.update_unread_counts(notifiable_unread_count, res.private_message_count);
```

Comment:
Hm, should this last block be a hook as well? I think `notifications.update_unread_counts` could have the `unread.calculate_notifiable_count` call inlined into it, or `unread.get_counts` could do that; maybe the latter makes more sense?